rust/src/measurements/make_private_expr/expr_index_candidates/mod.rs

```diff
@@ -72,117 +73,150 @@ static INDEX_CANDIDATES_PLUGIN: &str = "index_candidates";
+    }
+
+    fn call_udf(&self, _: &mut [Series]) -> PolarsResult<Option<Series>> {
+        polars_bail!(InvalidOperation: "OpenDP expressions must be passed through make_private_lazyframe to be executed.")
+    }
+
+    fn get_output(&self) -> Option<GetOutput> {
+        // dtype is unknown
+        Some(GetOutput::from_type(DataType::Null))
+    }
 }
 
 #[derive(Clone)]
 #[cfg_attr(feature = "ffi", derive(Deserialize, Serialize))]
-pub(crate) struct IndexCandidatesArgs {
-    pub candidates: Candidates,
+pub(crate) struct IndexCandidatesPlugin {
+    pub candidates: Series,
 }
 
-#[derive(Clone)]
-pub(crate) struct Candidates(pub Series);
+// #[derive(Clone)]
+// pub(crate) struct Candidates(pub Series);
 
-#[cfg(feature = "ffi")]
-impl Serialize for Candidates {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
-
-        for value in self.0.iter() {
-            match value {
-                AnyValue::Boolean(v) => seq.serialize_element(&v),
-                AnyValue::Int64(v) => seq.serialize_element(&v),
-                AnyValue::Float64(v) => seq.serialize_element(&v),
-                AnyValue::String(v) => seq.serialize_element(&v),
-                _ => Err(serde::ser::Error::custom(
-                    "Expected homogenous candidates of either bool, i64, f64, or string",
-                )),
-            }?;
-        }
-        seq.end()
-    }
-}
+// #[cfg(feature = "ffi")]
+// impl Serialize for Candidates {
+//     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+//     where
+//         S: serde::Serializer,
+//     {
+//         let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
 
-#[cfg(feature = "ffi")]
-impl<'de> Deserialize<'de> for Candidates {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        let value = Value::deserialize(deserializer)?;
-
-        let Value::List(list) = value else {
-            return Err(serde::de::Error::custom(format!(
-                "Expected a list, found {:?}",
-                value
-            )));
-        };
-        let first = list
-            .first()
-            .ok_or_else(|| serde::de::Error::custom("Expected at least one candidate"))?
-            .clone();
-
-        macro_rules! match_candidates {
-            ($list:ident, $dtype:ident) => {
-                Series::new(
-                    "",
-                    $list
-                        .into_iter()
-                        .map(|x| match x {
-                            Value::$dtype(v) => Ok(v),
-                            _ => Err(serde::de::Error::custom("Expected homogenous candidates")),
-                        })
-                        .collect::<Result<Vec<_>, D::Error>>()?,
-                )
-            };
-        }
+//         for value in self.0.iter() {
```

Comment:
Generally want to avoid commented out code blocks.

Reply:
Agree-- I wasn't sure if it would work without the code and forgot to return to it.